python/src/aiconfig/editor/client/src/components/prompt/prompt_input/schema_renderer/PromptInputAttachmentsSchemaRenderer.tsx

```diff
@@ -68,13 +68,21 @@ export default memo(function PromptInputAttachmentsSchemaRenderer({
     addedAttachments: InputAttachment[],
     index: number
   ) => {
-    const updatedAttachments = attachments.reduce((acc, attachment, i) => {
-      if (i === index) {
-        // Replace the attachment at the given index with the new attachments
-        return [...acc, ...addedAttachments];
-      }
-      return [...acc, attachment];
-    }, [] as InputAttachment[]);
+    let updatedAttachments;
+
+    // Add to end of attachments
+    if (index > attachments.length) {
+      updatedAttachments = [...attachments, ...addedAttachments];
+    } else {
+      updatedAttachments = attachments.reduce((acc, attachment, i) => {
+        if (i === index) {
+          // Replace the attachment at the given index with the new attachments
+          return [...acc, ...addedAttachments];
+        }
+        return [...acc, attachment];
+      }, [] as InputAttachment[]);
```

Comment:
Why not just do `attachments[index] = attachment` since we already know that `index <= attachments.length`?

Reply:
Hmm, good question. For some reason I was thinking it had to be a new array but would need to revisit later to see if that is the case